### PR TITLE
added auth context & login

### PR DESCRIPTION
**Pull request by tusharkandpal**
To Check:
✅ added `auth-context` & `authReducer`
✅ created Login page
✅added login & logout functionality
✅storing `encodedToken` & `user` object in localStorage

- created Wishlist component

### CODE REVIEW

File: src/context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { authInitialState, authReducer } from "../reducers/reducer";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, authInitialState);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (localStorage.getItem("encodedToken") !== null) {
```
Comment (JantuDeb) [inline src/context/auth-context.js]: You can do this in initial state if possible

**Review by JantuDeb — APPROVED** [review]

File: src/context/auth-context.js (hunk 1 above)
Comment (tusharkandpal) [reply src/context/auth-context.js]: yeah sure, will do. thanks!

**Review by psychoactiv — APPROVED** [review]
Good code  structure.

**Review by neetukumari4858 — APPROVED** [review]
code Is readable and good to see

**Pull request merged**